### PR TITLE
[Bug Fix] Attacks that miss against pokemon in semi invul state that have abilities such as volt absorb will not trigger while semi invulnerable

### PR DESCRIPTION
**Pull request by PrabbyDD**
## What are the changes the user will see?
This interaction will now work properly. For example, if thunderbolt is used against a pokemon with volt absorb, and that pokemon with volt absorb is in a semi invulnerable state, the attack will just miss instead of triggering the ability. 

related to #4337 

## Why am I making these changes?
Bug fix

## What are the changes from a developer perspective?
added check in move-effect-phase.ts for if pokemon is in semi invulnerable state

### Screenshots/Videos

https://github.com/user-attachments/assets/2d704a4f-2116-41cb-b91b-2278f09e9eb4

## How to test the changes?
Override files, run the updated volt absorb test file, and manually change return value of hitcheck() to false. 

## Checklist
- [X ] **I'm using `beta` as my base branch**
- [X ] There is no overlap with another PR?
- [ X] The PR is self-contained and cannot be split into smaller PRs?
- [X ] Have I provided a clear explanation of the changes?
- [X ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [X ] Have I tested the changes (manually)?
    - [X ] Are all unit tests still passing? (`npm run test`)
- [X ] Are the changes visual?
  - [X ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Snailman11** [comment]
@PrabbyDD Make sure to link your fixes with the bugs




Link to Original Discord Bug Report (aqua step (by quaq) heals wild politoed in bounce. If it can't be hit, shouldn't it not heal?)
https://discord.com/channels/1125469663833370665/1290401734593875969

**Comment by Tempo-anon** [comment]
what if the user or target have no guard?

File: src/phases/move-effect-phase.ts [hunk 1]
```diff
@@ -148,8 +149,9 @@ export class MoveEffectPhase extends PokemonPhase {
               && (hasConditionalProtectApplied.value || (!target.findTags(t => t instanceof DamageProtectedTag).length && target.findTags(t => t instanceof ProtectedTag).find(t => target.lapseTag(t.tagType)))
               || (this.move.getMove().category !== MoveCategory.STATUS && target.findTags(t => t instanceof DamageProtectedTag).find(t => target.lapseTag(t.tagType))));
 
-          /** Is the pokemon immune due to an ablility?  */
-          const isImmune = target.hasAbilityWithAttr(TypeImmunityAbAttr) && (target.getAbility()?.getAttrs(TypeImmunityAbAttr)?.[0]?.getImmuneType() === user.getMoveType(move));
+          /** Is the pokemon immune due to an ablility, and also not in a semi invulnerable state?  */
+          const isImmune = target.hasAbilityWithAttr(TypeImmunityAbAttr) && (target.getAbility()?.getAttrs(TypeImmunityAbAttr)?.[0]?.getImmuneType() === user.getMoveType(move))
+              && !targets[0].getTag(SemiInvulnerableTag);
```
Comment (DayKev) [inline src/phases/move-effect-phase.ts:154]: This one uses `target` and not `targets[0]` for the ability attr checks, shouldn't the tag check match that?

Comment (PrabbyDD) [reply src/phases/move-effect-phase.ts:154]: You are right, will submit an updated version

**Comment by PrabbyDD** [comment]
> what if the user or target have no guard?

No guard seemingly works fine. It bypasses the semi invul state, thus registering a hit, and triggering the ability.

**Pull request merged**